src/main/java/org/currencygoldexchangeapp/handlers/ExchangeRateAPIHandler.java

```diff
@@ -50,9 +51,11 @@ private HttpResponse<String> getHttpResponse(HttpRequest request) {
 
     private CurrencyExchange handleHttpResponse(HttpResponse<String> response) {
         if (response.statusCode() == HttpURLConnection.HTTP_OK) {
-            System.out.println("Status code: " + response.statusCode());
-            System.out.println("Response Body:");
+//            System.out.println("Status code: " + response.statusCode());
+//            System.out.println("Response Body:");
             return jsonMapper.deserializeJsonToCurrencyExchange(response.body());
+        } else if (response.statusCode() == HttpURLConnection.HTTP_NOT_FOUND) {
+            throw new CurrencyNotFoundException("The requested currency or date was not found.");
```

Comment:
CurrencyNotFound exception should only apply if the currency was not found, adding date here is confusing, don't you think?

Reply:
It's true. Exception has been changed to DataNotFoundException with proper message. Now CurrencyNotFoundException is used in another place to check properly if currency is available or not.